autorest/responder.go

```diff
@@ -142,6 +142,30 @@ func ByUnmarshallingJSON(v interface{}) RespondDecorator {
 	}
 }
 
+// ByUnmarshallingSuccessandErrorJSON returns a RespondDecorator that decodes a JSON document returned in the
+// response Body into the value pointed by success schema v or error schema e
+func ByUnmarshallingSuccessandErrorJSON(v interface{}, e interface{}) RespondDecorator {
```

Comment:
Disregard. The method needs to be renamed to `WithAADServiceErrorUnlessOK` and modified to unpack the body into an `AADServiceError` if the return code is not 200.